sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java

```diff
@@ -389,24 +418,32 @@ public void cancelled_analysis() {
     InputFile inputFile = inputFile(FILE_1);
     activeRules = (new ActiveRulesBuilder()).build();
     context.setCancelled(true);
-    sensor(null).execute(context);
+    sensor(null, null).execute(context);
     assertThat(context.measure(inputFile.key(), CoreMetrics.NCLOC)).isNull();
     assertThat(context.allAnalysisErrors()).isEmpty();
   }
 
   private PythonSensor sensor() {
-    return sensor(CUSTOM_RULES);
+    return sensor(CUSTOM_RULES, analysisWarning);
   }
 
-  private PythonSensor sensor(@Nullable PythonCustomRuleRepository[] customRuleRepositories) {
+  private PythonSensor sensor(@Nullable PythonCustomRuleRepository[] customRuleRepositories, @Nullable AnalysisWarnings analysisWarnings) {
     FileLinesContextFactory fileLinesContextFactory = mock(FileLinesContextFactory.class);
     FileLinesContext fileLinesContext = mock(FileLinesContext.class);
     when(fileLinesContextFactory.createFor(Mockito.any(InputFile.class))).thenReturn(fileLinesContext);
     CheckFactory checkFactory = new CheckFactory(activeRules);
     if(customRuleRepositories == null) {
-      return new PythonSensor(fileLinesContextFactory, checkFactory, new NoSonarFilter());
+      if (analysisWarnings == null) {
+        return new PythonSensor(fileLinesContextFactory, checkFactory, new NoSonarFilter());
+      } else {
+        return new PythonSensor(fileLinesContextFactory, checkFactory, new NoSonarFilter(), analysisWarnings);
```

Comment:
Why do we need this if/else? Can't we pass `analysisWarnings` as last argument even when it's null?